src/mlpack/methods/softmax_regression/softmax_regression.hpp

```diff
@@ -59,9 +59,7 @@ namespace regression {
  * regressor2.Classify(test_data, predictions2);
  * @endcode
  */
-template<
-  template<typename> class OptimizerType = mlpack::optimization::L_BFGS
->
+template<typename OptimizerType = mlpack::optimization::L_BFGS>
```

Comment:
I think we can templatize only `Train()` and the constructors that train the model, no need to store an optimizer internally.